<a href="https://colab.research.google.com/github/konkasivaji/690-stats-and-visulaization/blob/main/Assignment_04/assignment_04.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Reading the Data

In [30]:
import pandas as pd 
df = pd.read_csv("/content/MERGED2017_18_PP.csv")   # Reading the MERGED2017_18PP.scv data
df

/usr/local/lib/python3.7/dist-packages/IPython/core/interactiveshell.py:3326: DtypeWarning: Columns (1729) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


UNITID    OPEID  OPEID6  \
0       100654   100200    1002   
1       100663   105200    1052   
2       100690  2503400   25034   
3       100706   105500    1055   
4       100724   100500    1005   
...        ...      ...     ...   
7107  48312404   108163    1081   
7108  48387801  4223701   42237   
7109  48400201  4228101   42281   
7110  48778201  4221501   42215   
7111  49005401  4182601   41826   

                                                 INSTNM             CITY  \
0                              Alabama A & M University           Normal   
1                   University of Alabama at Birmingham       Birmingham   
2                                    Amridge University       Montgomery   
3                   University of Alabama in Huntsville       Huntsville   
4                              Alabama State University       Montgomery   
...                                                 ...              ...   
7107                   Arizona State University at Yuma             Yuma   
7108  Bay Area Medical Academy - San Jose Satellite ...         San Jose   
7109           High Desert Medical College - Bakerfield      Bakersfield   
7110                                       BCI - Malden           Malden   
7111               HCI College - Fort Lauderdale Campus  Fort Lauderdale   

     STABBR         ZIP  ACCREDAGENCY  INSTURL  NPCURL  ...  \
0        AL       35762           NaN      NaN     NaN  ...   
1        AL  35294-0110           NaN      NaN     NaN  ...   
2        AL  36117-3553           NaN      NaN     NaN  ...   
3        AL       35899           NaN      NaN     NaN  ...   
4        AL  36104-0271           NaN      NaN     NaN  ...   
...     ...         ...           ...      ...     ...  ...   
7107     AZ   853656900           NaN      NaN     NaN  ...   
7108     CA       95113           NaN      NaN     NaN  ...   
7109     CA       93301           NaN      NaN     NaN  ...   
7110     MA    21480000           NaN      NaN     NaN  ...   
7111     FL       33309           NaN      NaN     NaN  ...   

      OMENRUP_PARTTIME_POOLED_SUPP  FTFTPCTPELL  FTFTPCTFLOAN   UG12MN  \
0                              NaN       0.7106        0.7418   5207.0   
1                              NaN       0.4009        0.5488  13308.0   
2                              NaN       0.5000        0.5000    431.0   
3                              NaN       0.2012        0.4231   7519.0   
4                              NaN       0.7594        0.7402   5229.0   
...                            ...          ...           ...      ...   
7107                           NaN          NaN           NaN      NaN   
7108                           NaN          NaN           NaN      NaN   
7109                           NaN          NaN           NaN      NaN   
7110                           NaN          NaN           NaN      NaN   
7111                           NaN          NaN           NaN      NaN   

       G12MN  SCUGFFN  POOLYRS_FTFTAIDPCT  FTFTPCTPELL_POOLED_SUPP  \
0     1185.0   1410.0                 NaN                      NaN   
1     9888.0   1948.0                 NaN                      NaN   
2      437.0      4.0                 NaN                      NaN   
3     2378.0   1203.0                 NaN                      NaN   
4      687.0   1143.0                 NaN                      NaN   
...      ...      ...                 ...                      ...   
7107     NaN      NaN                 NaN                      NaN   
7108     NaN      NaN                 NaN                      NaN   
7109     NaN      NaN                 NaN                      NaN   
7110     NaN      NaN                 NaN                      NaN   
7111     NaN      NaN                 NaN                      NaN   

      FTFTPCTFLOAN_POOLED_SUPP  SCUGFFN_POOLED  
0                          NaN             NaN  
1                          NaN             NaN  
2                          NaN          

To Display How many rows and columns 

In [31]:
df.shape #To Count the number of rows and columns of Dataframe using df.shape

(7112, 1986)

1.  Number of colleges or rows are 7112 
2.  Number of variables or columns are 1986




To Display first 5 colleges

In [32]:
df.head() #This function is used to get to the first n rows of a dataframe.

UNITID    OPEID  OPEID6                               INSTNM        CITY  \
0  100654   100200    1002             Alabama A & M University      Normal   
1  100663   105200    1052  University of Alabama at Birmingham  Birmingham   
2  100690  2503400   25034                   Amridge University  Montgomery   
3  100706   105500    1055  University of Alabama in Huntsville  Huntsville   
4  100724   100500    1005             Alabama State University  Montgomery   

  STABBR         ZIP  ACCREDAGENCY  INSTURL  NPCURL  ...  \
0     AL       35762           NaN      NaN     NaN  ...   
1     AL  35294-0110           NaN      NaN     NaN  ...   
2     AL  36117-3553           NaN      NaN     NaN  ...   
3     AL       35899           NaN      NaN     NaN  ...   
4     AL  36104-0271           NaN      NaN     NaN  ...   

   OMENRUP_PARTTIME_POOLED_SUPP  FTFTPCTPELL  FTFTPCTFLOAN   UG12MN   G12MN  \
0                           NaN       0.7106        0.7418   5207.0  1185.0   
1                           NaN       0.4009        0.5488  13308.0  9888.0   
2                           NaN       0.5000        0.5000    431.0   437.0   
3                           NaN       0.2012        0.4231   7519.0  2378.0   
4                           NaN       0.7594        0.7402   5229.0   687.0   

   SCUGFFN  POOLYRS_FTFTAIDPCT  FTFTPCTPELL_POOLED_SUPP  \
0   1410.0                 NaN                      NaN   
1   1948.0                 NaN                      NaN   
2      4.0                 NaN                      NaN   
3   1203.0                 NaN                      NaN   
4   1143.0                 NaN                      NaN   

   FTFTPCTFLOAN_POOLED_SUPP  SCUGFFN_POOLED  
0                       NaN             NaN  
1                       NaN             NaN  
2                       NaN             NaN  
3                       NaN             NaN  
4                       NaN             NaN  

[5 rows x 1986 columns]

To Diaplay the last 5 colleges

In [33]:
df.tail() #This function is used to get to the last n rows of a dataframe.

UNITID    OPEID  OPEID6  \
7107  48312404   108163    1081   
7108  48387801  4223701   42237   
7109  48400201  4228101   42281   
7110  48778201  4221501   42215   
7111  49005401  4182601   41826   

                                                 INSTNM             CITY  \
7107                   Arizona State University at Yuma             Yuma   
7108  Bay Area Medical Academy - San Jose Satellite ...         San Jose   
7109           High Desert Medical College - Bakerfield      Bakersfield   
7110                                       BCI - Malden           Malden   
7111               HCI College - Fort Lauderdale Campus  Fort Lauderdale   

     STABBR        ZIP  ACCREDAGENCY  INSTURL  NPCURL  ...  \
7107     AZ  853656900           NaN      NaN     NaN  ...   
7108     CA      95113           NaN      NaN     NaN  ...   
7109     CA      93301           NaN      NaN     NaN  ...   
7110     MA   21480000           NaN      NaN     NaN  ...   
7111     FL      33309           NaN      NaN     NaN  ...   

      OMENRUP_PARTTIME_POOLED_SUPP  FTFTPCTPELL  FTFTPCTFLOAN  UG12MN  G12MN  \
7107                           NaN          NaN           NaN     NaN    NaN   
7108                           NaN          NaN           NaN     NaN    NaN   
7109                           NaN          NaN           NaN     NaN    NaN   
7110                           NaN          NaN           NaN     NaN    NaN   
7111                           NaN          NaN           NaN     NaN    NaN   

      SCUGFFN  POOLYRS_FTFTAIDPCT  FTFTPCTPELL_POOLED_SUPP  \
7107      NaN                 NaN                      NaN   
7108      NaN                 NaN                      NaN   
7109      NaN                 NaN                      NaN   
7110      NaN                 NaN                      NaN   
7111      NaN                 NaN                      NaN   

      FTFTPCTFLOAN_POOLED_SUPP  SCUGFFN_POOLED  
7107                       NaN             NaN  
7108                       NaN             NaN  
7109                       NaN             NaN  
7110                       NaN             NaN  
7111                       NaN             NaN  

[5 rows x 1986 columns]

To Display random 5 colleges

In [34]:
df.sample(5) #This function is used to get to the random  sample of items in a dataframe.

UNITID    OPEID  OPEID6                                     INSTNM  \
5294  449807  4036300   40363           Southeast Texas Career Institute   
1074  149727   693100    6931                Waubonsee Community College   
19    101286   101800    1018  George C Wallace Community College-Dothan   
4214  372718  2524000   25240                           Sunstate Academy   
1591  167215   891600    8916                     New England Law-Boston   

             CITY STABBR         ZIP  ACCREDAGENCY  INSTURL  NPCURL  ...  \
5294      Silsbee     TX       77656           NaN      NaN     NaN  ...   
1074  Sugar Grove     IL  60554-9454           NaN      NaN     NaN  ...   
19         Dothan     AL  36303-9234           NaN      NaN     NaN  ...   
4214   Clearwater     FL  33765-2702           NaN      NaN     NaN  ...   
1591       Boston     MA        2116           NaN      NaN     NaN  ...   

      OMENRUP_PARTTIME_POOLED_SUPP  FTFTPCTPELL  FTFTPCTFLOAN   UG12MN  G12MN  \
5294                           NaN       0.2708        0.1667    235.0    NaN   
1074                           NaN       0.3418        0.1543  18016.0    NaN   
19                             NaN       0.6521        0.0000   6098.0    NaN   
4214                           NaN       0.5952        0.8333    460.0    NaN   
1591                           NaN          NaN           NaN      NaN  622.0   

      SCUGFFN  POOLYRS_FTFTAIDPCT  FTFTPCTPELL_POOLED_SUPP  \
5294     48.0                 NaN                      NaN   
1074   1024.0                 NaN                      NaN   
19      733.0                 NaN                      NaN   
4214     42.0                 NaN                      NaN   
1591      NaN                 NaN                      NaN   

      FTFTPCTFLOAN_POOLED_SUPP  SCUGFFN_POOLED  
5294                       NaN             NaN  
1074                       NaN             NaN  
19                         NaN             NaN  
4214                       NaN             NaN  
1591                       NaN             NaN  

[5 rows x 1986 columns]

Create a smaller data frame with the following 9 variables:                                                                 
UNITID ,
INSTNM , 
CITY ,
STABBR ,
ZIP ,
ADM_RATE ,
UGDS ,
TUITIONFEE_IN ,
MN_EARN_WNE_P10

In [35]:
df_1 = df[['UNITID','INSTNM','CITY','STABBR','ZIP','ADM_RATE','UGDS','TUITIONFEE_IN','MN_EARN_WNE_P10']]
df_1

UNITID                                             INSTNM  \
0       100654                           Alabama A & M University   
1       100663                University of Alabama at Birmingham   
2       100690                                 Amridge University   
3       100706                University of Alabama in Huntsville   
4       100724                           Alabama State University   
...        ...                                                ...   
7107  48312404                   Arizona State University at Yuma   
7108  48387801  Bay Area Medical Academy - San Jose Satellite ...   
7109  48400201           High Desert Medical College - Bakerfield   
7110  48778201                                       BCI - Malden   
7111  49005401               HCI College - Fort Lauderdale Campus   

                 CITY STABBR         ZIP  ADM_RATE     UGDS  TUITIONFEE_IN  \
0              Normal     AL       35762    0.9027   4824.0         9857.0   
1          Birmingham     AL  35294-0110    0.9181  12866.0         8328.0   
2          Montgomery     AL  36117-3553       NaN    318.0         6900.0   
3          Huntsville     AL       35899    0.8123   6917.0        10280.0   
4          Montgomery     AL  36104-0271    0.9787   4189.0        11068.0   
...               ...    ...         ...       ...      ...            ...   
7107             Yuma     AZ   853656900       NaN      NaN            NaN   
7108         San Jose     CA       95113       NaN      NaN            NaN   
7109      Bakersfield     CA       93301       NaN      NaN            NaN   
7110           Malden     MA    21480000       NaN      NaN            NaN   
7111  Fort Lauderdale     FL       33309       NaN      NaN        22575.0   

      MN_EARN_WNE_P10  
0                 NaN  
1                 NaN  
2                 NaN  
3                 NaN  
4                 NaN  
...               ...  
7107              NaN  
7108              NaN  
7109              NaN  
7110              NaN  
7111              NaN  

[7112 rows x 9 columns]

Definition of the above 9 variables:


**Variables**
1.   UNITID: 
Unit ID for institution, which is a unique number given to each postsecondary institutions surveyed by IPEDS.

2.   LINSTNM: Institution name reported in IPEDS

3.   CITY: Location of the Institute

4.   STABBR: State postcode of the Institute

5.   ZIP: ZIP code of the Institute

6.   ADM_RATE: Accreditor for institution

7.   UGDS: consists of the number of degree/certificate-seeking undergraduates enrolled in the fall as reported by the IPEDS

8.   TUITIONFEE_IN: Fee structrue for Bachelor's degree in Personal And Culinary Services.

9.   MN_EARN_WNE_P10: The average salary is calculated for all undergraduates at an institution receiving federal financial aid in a given academic year who are also working full-time but not in school.







First 10 Colleges

In [36]:
df_1.head(10)

UNITID                               INSTNM            CITY STABBR  \
0  100654             Alabama A & M University          Normal     AL   
1  100663  University of Alabama at Birmingham      Birmingham     AL   
2  100690                   Amridge University      Montgomery     AL   
3  100706  University of Alabama in Huntsville      Huntsville     AL   
4  100724             Alabama State University      Montgomery     AL   
5  100751            The University of Alabama      Tuscaloosa     AL   
6  100760    Central Alabama Community College  Alexander City     AL   
7  100812              Athens State University          Athens     AL   
8  100830      Auburn University at Montgomery      Montgomery     AL   
9  100858                    Auburn University          Auburn     AL   

          ZIP  ADM_RATE     UGDS  TUITIONFEE_IN  MN_EARN_WNE_P10  
0       35762    0.9027   4824.0         9857.0              NaN  
1  35294-0110    0.9181  12866.0         8328.0              NaN  
2  36117-3553       NaN    318.0         6900.0              NaN  
3       35899    0.8123   6917.0        10280.0              NaN  
4  36104-0271    0.9787   4189.0        11068.0              NaN  
5  35487-0100    0.5330  32387.0        10780.0              NaN  
6       35010       NaN   1404.0         4440.0              NaN  
7       35611       NaN   2801.0            NaN              NaN  
8  36117-3596    0.8240   4211.0         8020.0              NaN  
9       36849    0.8393  23391.0        10968.0              NaN

Last 10 Colleges

In [37]:
df_1.tail(10)

UNITID                                             INSTNM  \
7102  48154401  National Personal Training Institute of Cleveland   
7103  48305401             Barber School of Pittsburgh - Ambridge   
7104  48312401                 Arizona State University at Tucson   
7105  48312402  Arizona State University - ASU Colleges at Lak...   
7106  48312403        Arizona State University at The Gila Valley   
7107  48312404                   Arizona State University at Yuma   
7108  48387801  Bay Area Medical Academy - San Jose Satellite ...   
7109  48400201           High Desert Medical College - Bakerfield   
7110  48778201                                       BCI - Malden   
7111  49005401               HCI College - Fort Lauderdale Campus   

                  CITY STABBR        ZIP  ADM_RATE  UGDS  TUITIONFEE_IN  \
7102  Highland Heights     OH      44143       NaN   NaN            NaN   
7103          Ambridge     PA  150030000       NaN   NaN            NaN   
7104            Tucson     AZ  857454284       NaN   NaN            NaN   
7105  Lake Havasu City     AZ  864036877       NaN   NaN         6904.0   
7106          Thatcher     AZ  855525545       NaN   NaN            NaN   
7107              Yuma     AZ  853656900       NaN   NaN            NaN   
7108          San Jose     CA      95113       NaN   NaN            NaN   
7109       Bakersfield     CA      93301       NaN   NaN            NaN   
7110            Malden     MA   21480000       NaN   NaN            NaN   
7111   Fort Lauderdale     FL      33309       NaN   NaN        22575.0   

      MN_EARN_WNE_P10  
7102              NaN  
7103              NaN  
7104              NaN  
7105              NaN  
7106              NaN  
7107              NaN  
7108              NaN  
7109              NaN  
7110              NaN  
7111              NaN

Random sample of 10 colleges

In [38]:
df_1.sample(10)

UNITID                                             INSTNM  \
2159    187912                      New Mexico Military Institute   
5025    443784                    Strayer University-Pennsylvania   
6915  21608301  South Hills School of Business and Technology-...   
3046    213570              Lincoln Technical Institute-Allentown   
1174    153418              Purdue University Global-Cedar Rapids   
6897  20627903                                  Sacramento Center   
4030    247700          National American University-Independence   
834     139940                           Georgia State University   
1342    157711                         Somerset Community College   
4442    407577              Scioto County Career Technical Center   

              CITY STABBR         ZIP  ADM_RATE     UGDS  TUITIONFEE_IN  \
2159       Roswell     NM  88201-5173    0.4226    419.0         4318.0   
5025       Trevose     PA       19053       NaN   1788.0        13857.0   
6915       Altoona     PA       16602       NaN      NaN        16608.0   
3046     Allentown     PA  18104-9889       NaN    444.0            NaN   
1174  Cedar Rapids     IA       52404       NaN    417.0        15352.0   
6897    Sacramento     CA   958341746       NaN      NaN        12896.0   
4030  Independence     MO  64057-1791       NaN    255.0        15000.0   
834        Atlanta     GA  30303-3083    0.6985  25031.0         9112.0   
1342      Somerset     KY  42501-2973       NaN   4435.0         4080.0   
4442    Lucasville     OH  45648-0766       NaN     81.0            NaN   

      MN_EARN_WNE_P10  
2159              NaN  
5025              NaN  
6915              NaN  
3046              NaN  
1174              NaN  
6897              NaN  
4030              NaN  
834               NaN  
1342              NaN  
4442              NaN

Find and display UMBC

In [39]:
df_2=df_1.loc[df_1['INSTNM'] == "University of Maryland-Baltimore County"]  # searching for umbc
df_2

UNITID                                   INSTNM       CITY STABBR  \
1478  163268  University of Maryland-Baltimore County  Baltimore     MD   

        ZIP  ADM_RATE     UGDS  TUITIONFEE_IN  MN_EARN_WNE_P10  
1478  21250    0.6021  11130.0        11518.0              NaN

Find Univ. of Maryland College Park

In [40]:
df_3=df_1.loc[df_1['INSTNM'] == "University of Maryland-College Park"] # 
df_3

UNITID                               INSTNM          CITY STABBR    ZIP  \
1479  163286  University of Maryland-College Park  College Park     MD  20742   

      ADM_RATE     UGDS  TUITIONFEE_IN  MN_EARN_WNE_P10  
1479     0.445  29273.0        10399.0              NaN

Find and display Loyola College Maryland

In [41]:
df_4=df_1.loc[df_1['INSTNM'] == "Loyola University Maryland"]   #
df_4

UNITID                      INSTNM       CITY STABBR         ZIP  \
1475  163046  Loyola University Maryland  Baltimore     MD  21210-2699   

      ADM_RATE    UGDS  TUITIONFEE_IN  MN_EARN_WNE_P10  
1475    0.7528  3903.0        47560.0              NaN

Find and display Johns Hopskins

In [42]:
df_5=df_1.loc[df_1['INSTNM'] == "Johns Hopkins University"]  # searching for john hopkins
df_5

UNITID                    INSTNM       CITY STABBR         ZIP  \
1473  162928  Johns Hopkins University  Baltimore     MD  21218-2688   

      ADM_RATE    UGDS  TUITIONFEE_IN  MN_EARN_WNE_P10  
1473    0.1254  5663.0        52170.0              NaN

Calculate the min, max, and average in-state tution of the above four institutions.

In [59]:
df_6 = pd.concat([df_2,df_3, df_4,df_5]) #Concating above 4 colleges
df_6

UNITID                                   INSTNM          CITY STABBR  \
1478  163268  University of Maryland-Baltimore County     Baltimore     MD   
1479  163286      University of Maryland-College Park  College Park     MD   
1475  163046               Loyola University Maryland     Baltimore     MD   
1473  162928                 Johns Hopkins University     Baltimore     MD   

             ZIP  ADM_RATE     UGDS  TUITIONFEE_IN  MN_EARN_WNE_P10  
1478       21250    0.6021  11130.0        11518.0              NaN  
1479       20742    0.4450  29273.0        10399.0              NaN  
1475  21210-2699    0.7528   3903.0        47560.0              NaN  
1473  21218-2688    0.1254   5663.0        52170.0              NaN

Caluculating Min

In [60]:
col_max = df_6['TUITIONFEE_IN'].max(axis = 0)
col_max

52170.0

Caluculating Min

In [61]:
col_min = df_6['TUITIONFEE_IN'].min(axis = 0)
col_min

10399.0

Caluculating Mean()

In [62]:
col_mean = df_6['TUITIONFEE_IN'].mean(axis = 0)
col_mean

30411.75

Filter the smaller dataset further to only keep colleges from Maryland, Virgina, Washington DC, Delaware, and Pennsylvinia. Save the filtered dataset to a new file named "Five_States_Colleges.csv" for future analysis. Upload this file to your GitHub along with the notebook.

In [63]:
t1=df.loc[df['STABBR'] == "MD"] # searching for colleges in maryland
t1

UNITID    OPEID  OPEID6                                  INSTNM  \
1446    161651  3097200   30972     Aesthetics Institute of Cosmetology   
1447    161688   205700    2057            Allegany College of Maryland   
1448    161767   205800    2058          Anne Arundel Community College   
1449    161776  1041000   10410               Brightwood College-Towson   
1450    161800   826300    8263                     Award Beauty School   
...        ...      ...     ...                                     ...   
6800  16210401   830801    8308          Cecil College - Elkton Station   
7004  43018401   145919    1459     Strayer University-Rockville Campus   
7005  43018402   145923    1459  Strayer University-Anne Arundel Campus   
7006  43018403   145924    1459   Strayer University-White Marsh Campus   
7007  43018404   145926    1459  Strayer University-Owings Mills Campus   

              CITY STABBR         ZIP  ACCREDAGENCY  INSTURL  NPCURL  ...  \
1446  Gaithersburg     MD       20877           NaN      NaN     NaN  ...   
1447    Cumberland     MD  21502-2596           NaN      NaN     NaN  ...   
1448        Arnold     MD  21012-1895           NaN      NaN     NaN  ...   
1449        Towson     MD  21286-2201           NaN      NaN     NaN  ...   
1450    Hagerstown     MD  21740-5610           NaN      NaN     NaN  ...   
...            ...    ...         ...           ...      ...     ...  ...   
6800        Elkton     MD   219215535           NaN      NaN     NaN  ...   
7004     Rockville     MD       20850           NaN      NaN     NaN  ...   
7005  Millersville     MD       21108           NaN      NaN     NaN  ...   
7006     Baltimore     MD       21236           NaN      NaN     NaN  ...   
7007  Owings Mills     MD       21117           NaN      NaN     NaN  ...   

      OMENRUP_PARTTIME_POOLED_SUPP  FTFTPCTPELL  FTFTPCTFLOAN   UG12MN  G12MN  \
1446                           NaN       0.8519        0.7778     76.0    NaN   
1447                           NaN       0.5991        0.5057   3471.0    NaN   
1448                           NaN       0.2411        0.1262  19908.0    NaN   
1449                           NaN       0.7694        0.7799    532.0    NaN   
1450                           NaN       0.9014        0.9155    302.0    NaN   
...                            ...          ...           ...      ...    ...   
6800                           NaN          NaN           NaN      NaN    NaN   
7004                           NaN          NaN           NaN      NaN    NaN   
7005                           NaN          NaN           NaN      NaN    NaN   
7006                           NaN          NaN           NaN      NaN    NaN   
7007                           NaN          NaN           NaN      NaN    NaN   

      SCUGFFN  POOLYRS_FTFTAIDPCT  FTFTPCTPELL_POOLED_SUPP  \
1446     27.0                 NaN                      NaN   
1447    439.0                 NaN                      NaN   
1448   1236.0                 NaN                      NaN   
1449    477.0                 NaN                      NaN   
1450    142.0                 NaN                      NaN   
...       ...                 ...                      ...   
6800      NaN                 NaN                      NaN   
7004      NaN                 NaN                      NaN   
7005      NaN                 NaN                      NaN   
7006      NaN                 NaN                      NaN   
7007      NaN                 NaN                      NaN   

      FTFTPCTFLOAN_POOLED_SUPP  SCUGFFN_POOLED  
1446                       NaN             NaN  
1447                       NaN             NaN  
1448                       NaN             NaN  
1449                       NaN             NaN  
1450                       NaN             NaN  
...                        ...             ...  
6800                       NaN             NaN  
7004                       NaN             NaN  
7005                   

In [49]:
t2=df.loc[df['STABBR'] == "VA"]   # Colleges in virgina Region

In [50]:
t3=df.loc[df['STABBR'] == "DC"]   ## Colleges in Washington DC
t3

UNITID     OPEID  OPEID6  \
643     131159    143400    1434   
644     131283    143700    1437   
645     131399    144100    1441   
646     131405   1280300   12803   
647     131450    144300    1443   
648     131469    144400    1444   
649     131496    144500    1445   
650     131520    144800    1448   
651     131803    145900    1459   
652     131830   2132800   21328   
653     131876    146000    1460   
654     131973    146400    1464   
4100    363721    144102    1441   
4348    384412   3218300   32183   
4616    420370   3104300   31043   
4844    439473   3409600   34096   
5079    444893   3750300   37503   
5328    450483  32098812   20988   
5520    455804   4114400   41144   
5521    455813   4142700   41427   
5739    459745   2155306   21553   
5747    459994    145900    1459   
6727  13180301    145904    1459   
7094  47503313   4208613   42086   

                                                 INSTNM        CITY STABBR  \
643                                 American University  Washington     DC   
644                  The Catholic University of America  Washington     DC   
645              University of the District of Columbia  Washington     DC   
646   Pontifical Faculty of the Immaculate Conceptio...  Washington     DC   
647                                Gallaudet University  Washington     DC   
648                        George Washington University  Washington     DC   
649                               Georgetown University  Washington     DC   
650                                   Howard University  Washington     DC   
651             Strayer University-District of Columbia  Washington     DC   
652              National Conservatory of Dramatic Arts  Washington     DC   
653                       Trinity Washington University  Washington     DC   
654                         Wesley Theological Seminary  Washington     DC   
4100  University of the District of Columbia-David A...  Washington     DC   
4348     University of the Potomac-Washington DC Campus  Washington     DC   
4616                         Career Technical Institute  Washington     DC   
4844                           Bennett Career Institute  Washington     DC   
5079                     Technical Learning Centers Inc  Washington     DC   
5328                University of Phoenix-Washington DC  Washington     DC   
5520                        Institute of World Politics  Washington     DC   
5521  Pontifical John Paul II Institute for Studies ...  Washington     DC   
5739  The Chicago School of Professional Psychology ...  Washington     DC   
5747                   Strayer University-Global Region  Washington     DC   
6727              Strayer University-Takoma Park Campus  Washington     DC   
7094  Relay Graduate School of Education - Washingto...  Washington     DC   

             ZIP  ACCREDAGENCY  INSTURL  NPCURL  ...  \
643   20016-8001           NaN      NaN     NaN  ...   
644        20064           NaN      NaN     NaN  ...   
645        20008           NaN      NaN     NaN  ...   
646   20017-1585           NaN      NaN     NaN  ...   
647   20002-3695           NaN      NaN     NaN  ...   
648        20052           NaN      NaN     NaN  ...   
649   20057-0001           NaN      NaN     NaN  ...   
650   20059-0001           NaN      NaN     NaN  ...   
651        20005           NaN      NaN     NaN  ...   
652        20007           NaN      NaN     NaN  ...   
653   20017-1094           NaN      NaN     NaN  ...   
654   20016-5690           NaN      NaN     NaN  ...   
4100       20008           NaN      NaN     NaN  ...   
4348       20005           NaN      NaN     NaN  ...   
4616       20005           NaN      NaN     NaN  ...   
4844       20017           NaN      NaN     NaN  ...   
5079  20006-5540           NaN      NaN     NaN  ...   
5328  20001-1431           NaN      NaN     NaN  ...   
5520       20036           NaN      NaN     NaN  ...   
5521       20064           NaN      

In [51]:
t4=df.loc[df['STABBR'] == "PA"]   # Colleges in Pennsylvinia
t4

UNITID    OPEID  OPEID6  \
2933    210474  2178900   21789   
2934    210492   322800    3228   
2935    210508  3353300   33533   
2936    210571   322900    3229   
2937    210599  2495500   24955   
...        ...      ...     ...   
7023  44378405   145950    1459   
7024  44378406   145976    1459   
7063  45794301  4149801   41498   
7089  47503308  4208608   42086   
7103  48305401  4224601   42246   

                                                 INSTNM           CITY STABBR  \
2933             Jolie Hair and Beauty Academy-Hazleton       Hazleton     PA   
2934               Bryn Athyn College of the New Church     Bryn Athyn     PA   
2935                              Academy of Vocal Arts   Philadelphia     PA   
2936                                   Albright College        Reading     PA   
2937                            All-State Career School      Essington     PA   
...                                                 ...            ...    ...   
7023               Strayer University-Warrendale Campus     Warrendale     PA   
7024                Strayer University-Allentown Campus  Center Valley     PA   
7063   Somerset County Technology Center - Bedford Site        Bedford     PA   
7089  Relay Graduate School of Education - Philadelp...   Philadelphia     PA   
7103             Barber School of Pittsburgh - Ambridge       Ambridge     PA   

             ZIP  ACCREDAGENCY  INSTURL  NPCURL  ...  \
2933       18201           NaN      NaN     NaN  ...   
2934  19009-0711           NaN      NaN     NaN  ...   
2935  19103-6685           NaN      NaN     NaN  ...   
2936  19612-5234           NaN      NaN     NaN  ...   
2937       19029           NaN      NaN     NaN  ...   
...          ...           ...      ...     ...  ...   
7023       15086           NaN      NaN     NaN  ...   
7024       18034           NaN      NaN     NaN  ...   
7063       15522           NaN      NaN     NaN  ...   
7089   191224906           NaN      NaN     NaN  ...   
7103   150030000           NaN      NaN     NaN  ...   

      OMENRUP_PARTTIME_POOLED_SUPP  FTFTPCTPELL  FTFTPCTFLOAN  UG12MN  G12MN  \
2933                           NaN       0.7838        0.7027   116.0    NaN   
2934                           NaN       0.4857        0.6571   359.0    7.0   
2935                           NaN          NaN           NaN     NaN   31.0   
2936                           NaN       0.5252        0.8674  2433.0   26.0   
2937                           NaN       0.7548        0.7194   608.0    NaN   
...                            ...          ...           ...     ...    ...   
7023                           NaN          NaN           NaN     NaN    NaN   
7024                           NaN          NaN           NaN     NaN    NaN   
7063                           NaN          NaN           NaN     NaN    NaN   
7089                           NaN          NaN           NaN     NaN    NaN   
7103                           NaN          NaN           NaN     NaN    NaN   

      SCUGFFN  POOLYRS_FTFTAIDPCT  FTFTPCTPELL_POOLED_SUPP  \
2933     74.0                 NaN                      NaN   
2934     70.0                 NaN                      NaN   
2935      NaN                 NaN                      NaN   
2936    596.0                 NaN                      NaN   
2937    310.0                 NaN                      NaN   
...       ...                 ...                      ...   
7023      NaN                 NaN                      NaN   
7024      NaN                 NaN                      NaN   
7063      NaN                 NaN                      NaN   
7089      NaN                 NaN                      NaN   
7103      NaN                 NaN                      NaN   

      FTFTPCTFLOAN_POOLED_SUPP  SCUGFFN_POOLED  
2933                       NaN             NaN  
2934                       NaN             NaN  
2935                       NaN             NaN  
2936                       NaN         

In [52]:
t5=df.loc[df['STABBR'] == "DE"]   # Colleges in Delaware
t5

UNITID    OPEID  OPEID6  \
634     130837  2125200   21252   
635     130873  3025800   30258   
636     130907  1172700   11727   
637     130934   142800    1428   
638     130943   143100    1431   
639     130989   142900    1429   
640     131061  2190400   21904   
641     131098   143300    1433   
642     131113   794800    7948   
4729    432524  4139800   41398   
4917    441362  3543300   35433   
5037    444200  2586902   25869   
5275    449472  3702300   37023   
5322    450298   145948    1459   
5354    451051  4130700   41307   
5718    459301  4171100   41711   
6527    488262  4250300   42503   
6725  13090701  1172702   11727   
6726  13090702  1172703   11727   
7084  47503303  4208605   42086   

                                                 INSTNM        CITY STABBR  \
634   Margaret H Rollins School of Nursing at Beebe ...       Lewes     DE   
635                           Dawn Career Institute LLC      Newark     DE   
636          Delaware Technical Community College-Terry       Dover     DE   
637                           Delaware State University       Dover     DE   
638                              University of Delaware      Newark     DE   
639                               Goldey-Beacom College  Wilmington     DE   
640         Schilling-Douglas School of Hair Design LLC      Newark     DE   
641                                      Wesley College       Dover     DE   
642                               Wilmington University  New Castle     DE   
4729                 Delaware College of Art and Design  Wilmington     DE   
4917        Harris School of Business-Wilmington Campus  Wilmington     DE   
5037             Harris School of Business-Dover Campus       Dover     DE   
5275         Delaware Learning Institute of Cosmetology    Dagsboro     DE   
5322                        Strayer University-Delaware      Newark     DE   
5354                    Academy of Massage and Bodywork        Bear     DE   
5718                  Paul Mitchell the School-Delaware      Newark     DE   
6527          Hair Academy School of Barbering & Beauty      Newark     DE   
6725         Delaware Technical Community College-Owens  Georgetown     DE   
6726  Delaware Technical Community College-Stanton/W...  Wilmington     DE   
7084      Relay Graduate School of Education - Delaware  Wilmington     DE   

             ZIP  ACCREDAGENCY  INSTURL  NPCURL  ...  \
634        19958           NaN      NaN     NaN  ...   
635        19702           NaN      NaN     NaN  ...   
636        19901           NaN      NaN     NaN  ...   
637        19901           NaN      NaN     NaN  ...   
638        19716           NaN      NaN     NaN  ...   
639        19808           NaN      NaN     NaN  ...   
640        19711           NaN      NaN     NaN  ...   
641   19901-3875           NaN      NaN     NaN  ...   
642        19720           NaN      NaN     NaN  ...   
4729  19801-3007           NaN      NaN     NaN  ...   
4917       19803           NaN      NaN     NaN  ...   
5037       19901           NaN      NaN     NaN  ...   
5275       19939           NaN      NaN     NaN  ...   
5322       19713           NaN      NaN     NaN  ...   
5354       19701           NaN      NaN     NaN  ...   
5718       19702           NaN      NaN     NaN  ...   
6527  19713-2305           NaN      NaN     NaN  ...   
6725       19947           NaN      NaN     NaN  ...   
6726       19801           NaN      NaN     NaN  ...   
7084       19801           NaN      NaN     NaN  ...   

      OMENRUP_PARTTIME_POOLED_SUPP  FTFTPCTPELL  FTFTPCTFLOAN   UG12MN  \
634                            NaN          NaN           NaN     53.0   
635                            NaN       0.7108        0.7292    627.0   
636                            NaN       0.5172        0.0923  19385.0   
637                            NaN       0.5579        0.7524   4770.0   
638                            NaN       0.1391        0.5037  20550.0   
639         

In [54]:

 FiveStatesColleges= pd.concat([t1,t2,t3,t4,t5], ignore_index=True)  #Merging all the required filtered states from data set and make it to one data frame
 FiveStatesColleges


UNITID    OPEID  OPEID6  \
0      161651  3097200   30972   
1      161688   205700    2057   
2      161767   205800    2058   
3      161776  1041000   10410   
4      161800   826300    8263   
..        ...      ...     ...   
685    459301  4171100   41711   
686    488262  4250300   42503   
687  13090701  1172702   11727   
688  13090702  1172703   11727   
689  47503303  4208605   42086   

                                                INSTNM          CITY STABBR  \
0                  Aesthetics Institute of Cosmetology  Gaithersburg     MD   
1                         Allegany College of Maryland    Cumberland     MD   
2                       Anne Arundel Community College        Arnold     MD   
3                            Brightwood College-Towson        Towson     MD   
4                                  Award Beauty School    Hagerstown     MD   
..                                                 ...           ...    ...   
685                  Paul Mitchell the School-Delaware        Newark     DE   
686          Hair Academy School of Barbering & Beauty        Newark     DE   
687         Delaware Technical Community College-Owens    Georgetown     DE   
688  Delaware Technical Community College-Stanton/W...    Wilmington     DE   
689      Relay Graduate School of Education - Delaware    Wilmington     DE   

            ZIP  ACCREDAGENCY  INSTURL  NPCURL  ...  \
0         20877           NaN      NaN     NaN  ...   
1    21502-2596           NaN      NaN     NaN  ...   
2    21012-1895           NaN      NaN     NaN  ...   
3    21286-2201           NaN      NaN     NaN  ...   
4    21740-5610           NaN      NaN     NaN  ...   
..          ...           ...      ...     ...  ...   
685       19702           NaN      NaN     NaN  ...   
686  19713-2305           NaN      NaN     NaN  ...   
687       19947           NaN      NaN     NaN  ...   
688       19801           NaN      NaN     NaN  ...   
689       19801           NaN      NaN     NaN  ...   

     OMENRUP_PARTTIME_POOLED_SUPP  FTFTPCTPELL  FTFTPCTFLOAN   UG12MN  G12MN  \
0                             NaN       0.8519        0.7778     76.0    NaN   
1                             NaN       0.5991        0.5057   3471.0    NaN   
2                             NaN       0.2411        0.1262  19908.0    NaN   
3                             NaN       0.7694        0.7799    532.0    NaN   
4                             NaN       0.9014        0.9155    302.0    NaN   
..                            ...          ...           ...      ...    ...   
685                           NaN       0.5641        0.9231    179.0    NaN   
686                           NaN       0.5556        0.5556     46.0    NaN   
687                           NaN          NaN           NaN      NaN    NaN   
688                           NaN          NaN           NaN      NaN    NaN   
689                           NaN          NaN           NaN      NaN    NaN   

     SCUGFFN  POOLYRS_FTFTAIDPCT  FTFTPCTPELL_POOLED_SUPP  \
0       27.0                 NaN                      NaN   
1      439.0                 NaN                      NaN   
2     1236.0                 NaN                      NaN   
3      477.0                 NaN                      NaN   
4      142.0                 NaN                      NaN   
..       ...                 ...                      ...   
685     39.0                 NaN                      NaN   
686     27.0                 NaN                      NaN   
687      NaN                 NaN                      NaN   
688      NaN                 NaN                      NaN   
689      NaN                 NaN                      NaN   

     FTFTPCTFLOAN_POOLED_SUPP  SCUGFFN_POOLED  
0                         NaN             NaN  
1                         NaN             NaN  
2                         NaN             NaN  
3                         NaN             NaN  
4                         NaN             NaN  
..    

In [56]:
FiveStatesColleges.to_csv('Five_States_Colleges.csv', index=False) ## converting data frame to CSV File


In [58]:
l1 = pd.read_csv('Five_States_Colleges.csv')
l1

UNITID    OPEID  OPEID6  \
0      161651  3097200   30972   
1      161688   205700    2057   
2      161767   205800    2058   
3      161776  1041000   10410   
4      161800   826300    8263   
..        ...      ...     ...   
685    459301  4171100   41711   
686    488262  4250300   42503   
687  13090701  1172702   11727   
688  13090702  1172703   11727   
689  47503303  4208605   42086   

                                                INSTNM          CITY STABBR  \
0                  Aesthetics Institute of Cosmetology  Gaithersburg     MD   
1                         Allegany College of Maryland    Cumberland     MD   
2                       Anne Arundel Community College        Arnold     MD   
3                            Brightwood College-Towson        Towson     MD   
4                                  Award Beauty School    Hagerstown     MD   
..                                                 ...           ...    ...   
685                  Paul Mitchell the School-Delaware        Newark     DE   
686          Hair Academy School of Barbering & Beauty        Newark     DE   
687         Delaware Technical Community College-Owens    Georgetown     DE   
688  Delaware Technical Community College-Stanton/W...    Wilmington     DE   
689      Relay Graduate School of Education - Delaware    Wilmington     DE   

            ZIP  ACCREDAGENCY  INSTURL  NPCURL  ...  \
0         20877           NaN      NaN     NaN  ...   
1    21502-2596           NaN      NaN     NaN  ...   
2    21012-1895           NaN      NaN     NaN  ...   
3    21286-2201           NaN      NaN     NaN  ...   
4    21740-5610           NaN      NaN     NaN  ...   
..          ...           ...      ...     ...  ...   
685       19702           NaN      NaN     NaN  ...   
686  19713-2305           NaN      NaN     NaN  ...   
687       19947           NaN      NaN     NaN  ...   
688       19801           NaN      NaN     NaN  ...   
689       19801           NaN      NaN     NaN  ...   

     OMENRUP_PARTTIME_POOLED_SUPP  FTFTPCTPELL  FTFTPCTFLOAN   UG12MN  G12MN  \
0                             NaN       0.8519        0.7778     76.0    NaN   
1                             NaN       0.5991        0.5057   3471.0    NaN   
2                             NaN       0.2411        0.1262  19908.0    NaN   
3                             NaN       0.7694        0.7799    532.0    NaN   
4                             NaN       0.9014        0.9155    302.0    NaN   
..                            ...          ...           ...      ...    ...   
685                           NaN       0.5641        0.9231    179.0    NaN   
686                           NaN       0.5556        0.5556     46.0    NaN   
687                           NaN          NaN           NaN      NaN    NaN   
688                           NaN          NaN           NaN      NaN    NaN   
689                           NaN          NaN           NaN      NaN    NaN   

     SCUGFFN  POOLYRS_FTFTAIDPCT  FTFTPCTPELL_POOLED_SUPP  \
0       27.0                 NaN                      NaN   
1      439.0                 NaN                      NaN   
2     1236.0                 NaN                      NaN   
3      477.0                 NaN                      NaN   
4      142.0                 NaN                      NaN   
..       ...                 ...                      ...   
685     39.0                 NaN                      NaN   
686     27.0                 NaN                      NaN   
687      NaN                 NaN                      NaN   
688      NaN                 NaN                      NaN   
689      NaN                 NaN                      NaN   

     FTFTPCTFLOAN_POOLED_SUPP  SCUGFFN_POOLED  
0                         NaN             NaN  
1                         NaN             NaN  
2                         NaN             NaN  
3                         NaN             NaN  
4                         NaN             NaN  
..    